app/pages/change_password/change_password.jsx

```diff
@@ -0,0 +1,278 @@
+import React, { useState, useEffect } from 'react';
+import { post } from '../../services/apiServices';
+import { IncorrectCredentials, StatusBlock } from '../../components';
+import { modalTypes, toastMode } from '../../utils/constants';
+import './change_password.less';
+import { useToast } from '../../components/toast_provider';
+
+const ChangePassword = () => {
+  const { addToast } = useToast()
+  const [status, setStatus] = useState(modalTypes.default);
+  const [incorrectCredentials, setIncorrectCredentials] = useState(false);
+  const [showPass, setShowPassValue] = useState({
+    password: false,
+    confirmPassword: false,
+  });
+  const [newPassValue, setNewPassValue] = useState(
+    sessionStorage.getItem('new_pass') || ''
+  );
+  const [newValidPass, setNewValidPass] = useState(true);
+  const [confirmPassValue, setConfirmPassValue] = useState(
+    sessionStorage.getItem('confirm_pass') || ''
+  );
+  const [confirmValidPass, setConfirmValidPass] = useState(true);
+
+  const handleSetNewPassValue = () => {
+    if (newPassValue?.trim().length > 3) {
+      setNewValidPass(true);
+      return true;
+    }
+    setNewValidPass(false);
+    return false;
+  };
+  const handleSetConfirmPassValue = () => {
+    if (confirmPassValue?.trim().length > 3) {
+      setConfirmValidPass(true);
+      return true;
+    }
+    setNewValidPass(false);
+    return false;
+  };
+  const handleNewChangePass = (e) => {
+    const value = e.target.value.trim();
+    setNewPassValue(value);
+    setNewValidPass(true);
+    sessionStorage.setItem('new_pass', value);
+  };
+  const handleConfirmChangePass = (e) => {
+    const value = e.target.value.trim();
+    setConfirmPassValue(value);
+    setConfirmValidPass(true);
+    sessionStorage.setItem('confirm_pass', value);
+  };
+
+  const handleSubmit = () => {
+    if ((newPassValue && newPassValue.length < 3) || !newPassValue) {
+      setNewValidPass(false);
+    }
+    if (
+      (confirmPassValue && confirmPassValue.length < 3)
+      || !confirmPassValue
+    ) {
+      setConfirmValidPass(false);
+    }
+    if (confirmValidPass !== newValidPass) {
+      setConfirmPassValue(false);
+    }
+  };
+
+  const handleFormSubmit = (e) => {
+    e.preventDefault();
+    if (
+      (newPassValue === confirmPassValue)
+        && ((newPassValue && newPassValue.length > 3) || newPassValue)
```

Comment:
This condition looks complicated 
Why you need the `|| newPassValue` and `|| confirmPassValue`
Maybe to divide it to some boolean vars with readable names?